components/frontend/src/header_footer/ResetSettingsButton.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Icon } from 'semantic-ui-react';
+import { Popup } from '../semantic_ui_react_wrappers';
+import { optionalDatePropType, settingsPropType } from '../sharedPropTypes';
+
+export function ResetSettingsButton({ atReportsOverview, handleDateChange, reportDate, settings }) {
+    const metricsToHideDefault = atReportsOverview ? "all" : "none"
+    const label = `Reset ${atReportsOverview ? "reports overview" : "this report's"} settings`
+    return (
+        <Popup
+            on={["hover", "focus"]}
+            trigger={
+                <span  // We need a span here to prevent the popup from becoming disabled whenever the button is disabled
+                >
+                    <Button
+                        aria-label={label}
+                        basic
+                        disabled={
+                            settings.dateInterval.isDefault() &&
+                            settings.dateOrder.isDefault() &&
+                            settings.hiddenCards.isDefault() &&
+                            settings.hiddenColumns.isDefault() &&
+                            settings.hiddenTags.isDefault() &&
+                            settings.metricsToHide.equals(metricsToHideDefault) &&
+                            settings.nrDates.isDefault() &&
+                            settings.showIssueCreationDate.isDefault() &&
+                            settings.showIssueDueDate.isDefault() &&
+                            settings.showIssueRelease.isDefault() &&
+                            settings.showIssueSprint.isDefault() &&
+                            settings.showIssueSummary.isDefault() &&
+                            settings.showIssueUpdateDate.isDefault() &&
+                            settings.sortColumn.isDefault() &&
+                            settings.sortDirection.isDefault() &&
+                            settings.expandedItems.isDefault() &&
+                            reportDate === null
+                        }
+                        icon
+                        onClick={() => {
+                            handleDateChange(null);
+                            settings.dateInterval.reset();
```

Comment:
Why not create a `settings.reset()` which does it all?

Reply:
Yeah, I thought about it and then forgot. Will add it. And a `settings.areAllDefault()` method.